ROMFS/px4fmu_common/init.d/rc.mavlink

```diff
@@ -0,0 +1,255 @@
+#!nsh
+#
+# MAVLINK stream startup script.
+#
+# NOTE: environment variable references:
+#    If the dollar sign ('$') is followed by a left bracket ('{') then the
+#    variable name is terminated with the right bracket character ('}').
+#    Otherwise, the variable name goes to the end of the argument.
+#
+#
+# NOTE: COMMENT LINES ARE REMOVED BEFORE STORED IN ROMFS.
+#
+# NOTE: UART Mapping is documented in the rcS script and variables
+#       are declared/initialized/set/unset in the rcS script.
+#
+# NOTE: Normal mode uses baud rate of 57600 (default) and data rate of 1000 bytes/s.
+#
+# NOTE: To utilize this startup script, set mavlink device stream settings in the
+#       respective airframe file as shown here:
+#
+#       set TELEMETRY_DEVICE_2          "/dev/ttyS2"
+#       set TELEMETRY_DEVICE_2_ARGS     "-m onboard -r 10000 -x -f"
+#       set TELEMETRY_DEVICE_2_BAUD     "57600"
+#
+
+echo "Executing rc.mavlink"
+
+#
+# Declare local script variables and initialize default values
+#
+
+
+#
+# Setup for board specific configurations
+#
+if ver hwcmp AEROFC_V1
+then
+	set TELEMETRY_DEVICE_1      "/dev/ttyS3"
+	set TELEMETRY_DEVICE_1_ARGS " -r 1200"
+
+	# Only start mavlink if the Benewake TFMini or LeddarOne isn't being used
+	if param greater SENS_EN_TFMINI 0
+	then
+		set TELEMETRY_DEVICE_1 none
+	fi
+
+	if param greater SENS_EN_LEDDAR1 0
+	then
+		set TELEMETRY_DEVICE_1 none
+	fi
+
+	if protocol_splitter start ${TELEMETRY_DEVICE_2}
+	then
+		mavlink start -d /dev/mavlink -b 921600 -m onboard -r 5000 -x
+		micrortps_client start -d /dev/rtps -b 921600 -l -1 -s 2000
+	fi
+fi
+
+if ver hwcmp CRAZYFLIE
+then
+	# Avoid using either of the two available serial devices
+	set TELEMETRY_DEVICE_1 none
+fi
+
+if ver hwcmp NXPHLITE_V3
+then
+	set TELEMETRY_DEVICE_2 "/dev/ttyS4"
+fi
+
+if ver hwcmp MINDPX_V2 PX4FMU_V4 PX4FMU_V4PRO
+then
+	# This is TELEM4 on Pixhawk 3 Pro
+	frsky_telemetry start -d /dev/ttyS6
+
+	if ver hwcmp PX4FMU_V4
+	then
+		# Setup MAVLink on Wifi (ESP8266 port)
+		set TELEMETRY_DEVICE_0_ARGS "-r 20000"
+		set TELEMETRY_DEVICE_0_BAUD "921600"
+
+		# Use ttyS1 for MAVLink on FMUv4 in addition to ttyS0 (debug)
+		set TELEMETRY_DEVICE_1_ARGS "-r 1200"
+	fi
+fi
+
+#
+# MAVLink onboard / TELEM2
+#
+if param compare SYS_COMPANION 10
+then
+	frsky_telemetry start -d ${TELEMETRY_DEVICE_2}
+fi
+
+if param compare SYS_COMPANION 20
```

Comment:
Note that we plan to completely generalize UART config (https://github.com/PX4/Firmware/issues/8302), and `SYS_COMPANION` will probably fall away then.

Reply:
Agreed.  I think this is a good plan.  The airframe config files are already modified in this PR such that they no longer utilize SYS_COMPANION so deprecating that param will be simple if this PR is accepted.